sympy/core/mul.py

```diff
@@ -1222,7 +1222,10 @@ def _eval_is_prime(self):
             r = S.One
             for arg in self.args:
                 r *= arg
-            return r.is_prime
+            if self != r:
+                return r.is_prime
+            else:
+                return None
```

Comment:
Oh, and use `is` instead of `==`/`!=`. The latter calls into SymPy's `__eq__`, so it could easily run into the same sort of cycle.